OneCommunityGlobal/HighestGoodNetworkApp
# Akshay Fix Weekly Summary Tab Dates

# Description
This PR fixes the date calculation issue in the WeeklySummary component that caused an 8-day gap between the 'Last Week' and 'Week Before Last' tabs. The issue was due to incorrect date calculations in the moment.js library, caused by Daylight Savings Time (DST).
Fixes Weekly Summaries Component (Priority High) bug

## Mainly changes explained:
Update date calculation logic in WeeklySummary.jsx to correctly calculate the due dates for the 'Last Week', 'Week Before Last', and 'Three Weeks Ago' tabs.
Add comments to the code to explain the date calculation logic.

## How to test:
1. Check out the current branch.
2. Run npm install to ensure all dependencies are installed.
3. Log in as an admin user.
4. Go to the Weekly Summary component by clicking on Summary in the Dashboard.
5. Verify that the dates on the 'Last Week', 'Week Before Last', and 'Three Weeks Ago' tabs are calculated correctly and there is no 8-day gap between them.

## Screenshots or videos of changes:
Before Fix:
<img width="797" alt="screenshot4" src="https://user-images.githubusercontent.com/94432002/227760609-b74b4969-b2c7-4b3f-9207-365991e5dad5.png">
After Fix:
<img width="794" alt="screenshot3" src="https://user-images.githubusercontent.com/94432002/227760646-d836bf86-77a0-44d4-8512-fff6d8c99a80.png">

Note:
This PR includes only the changes related to fixing the date calculation issue in the WeeklySummary component. The code has been tested locally, and the issue has been resolved.
